packages/tao/src/commands/migrate.ts

```diff
@@ -439,7 +447,23 @@ function createFetcher() {
     packageVersion: string
   ): Promise<MigrationsJson> {
     if (!cache[`${packageName}-${packageVersion}`]) {
-      const dir = dirSync().name;
+      // Setting up a tmpDir. In order to obey various .npmrc files
+      // this is a bit more complex than it should be. `.npmrc` can be present
+      // in ~/, workspace root, global, or in the npm installation directory.
+      const tmpdir = `${appRootPath}/node_modules/.tmp`;
+      mkdirSync(tmpdir, { recursive: true }); // dirSync fails if tmpdir does not exist yet.
+      const dir = dirSync({
+        tmpdir: tmpdir,
+      }).name;
+      // Creating a package.json in the new tmp directory prevents us from
+      // installing packages in the actual workspace instead of this subdirectory.
+      writeFileSync(`${dir}/package.json`, '{}');
+      // Copy npmrc from workspace root if it exists, so that npm still follows it.
+      const npmrcPath = `${appRootPath}/.npmrc`;
+      if (existsSync(npmrcPath)) {
+        copyFileSync(npmrcPath, `${dir}/.npmrc`);
```

Comment:
This is inside of the workspace so it should automatically be inherited?

Reply:
.npmrc does not inherit in this way (on windows for sure, but I think thats the case everywhere). I tried without it.

Reply:
https://docs.npmjs.com/cli/v7/configuring-npm/npmrc describes there only being 4 files loaded, in order to directly load the workspace .npmrc we would need to not have a package.json, but then package installation would not hit the temp directory